Patch-Source: https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=181aa56d715851ed0cd750a1c503ff26826766ca
--
From 181aa56d715851ed0cd750a1c503ff26826766ca Mon Sep 17 00:00:00 2001
From: Marek Szuba <marecki@gentoo.org>
Date: Tue, 30 Jan 2024 11:07:29 +0000
Subject: media-gfx/pngcrush: only use PNG_IGNORE_ADLER32 if supported

Since libpng-1.6.41 support for PNG_IGNORE_ADLER32 is optional, disabled
by default owing to its use of an undocumented zlib function, and not
presently enabled in Gentoo.

Funnily enough there are *two* places in pngcrush code where this option
is used and one of them has already used an ifdef guard.

Closes: https://bugs.gentoo.org/923023


--- a/pngcrush.c
+++ b/pngcrush.c
@@ -5520,8 +5520,10 @@
                      * they were already checked in the pngcrush_measure_idat
                      * function
                      */
+# ifdef PNG_IGNORE_ADLER32
                     png_set_option(read_ptr, PNG_IGNORE_ADLER32,
                         PNG_OPTION_ON);
+# endif
                     png_set_crc_action(read_ptr, PNG_CRC_QUIET_USE,
                                        PNG_CRC_QUIET_USE);
                 }
